src/main/java/org/folio/exceptions/ExceptionUtil.java

```diff
@@ -0,0 +1,59 @@
+package org.folio.exceptions;
+
+import io.vertx.core.json.JsonObject;
+import org.apache.commons.lang3.StringUtils;
+import org.folio.invoices.rest.exceptions.HttpException;
+import org.folio.rest.jaxrs.model.Error;
+import org.folio.rest.jaxrs.model.Errors;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static java.util.stream.Collectors.toList;
+import static org.folio.invoices.utils.ErrorCodes.GENERIC_ERROR_CODE;
+
+public class ExceptionUtil {
+  private static final String ERROR_CAUSE = "cause";
+
+  private ExceptionUtil() {
+  }
+
+  public static Errors convertToErrors(Throwable throwable) {
+    final Throwable cause = Optional.ofNullable(throwable.getCause()).orElse(throwable);
+    Errors errors;
+    if (cause instanceof HttpException) {
+      errors = ((HttpException) cause).getErrors();
+      List<Error> errorList = errors.getErrors().stream().map(ExceptionUtil::mapToError).collect(toList());
+      errors.setErrors(errorList);
+    } else {
+      errors = new Errors().withErrors(Collections.singletonList(GENERIC_ERROR_CODE.toError()
+                           .withAdditionalProperty(ERROR_CAUSE, cause.getMessage())))
+                           .withTotalRecords(1);
+    }
+    return errors;
+  }
+
+  @SuppressWarnings("java:S5852")
+  public static boolean isErrorMessageJson(String errorMessage) {
```

Comment:
please add a comment here describing what the method does, example input, etc.

Reply:
i have updated the logic for error  response building and removed this entire class , requesting you to please review